Dev

AlejoART opened the pull request:
New PR ready for revision.

alejoramtor committed:
first commit

alejoramtor committed:
Setting the base POM class with the general driver methods

alejoramtor committed:
This is the first class for selecting the Santander league

alejoramtor committed:
This page is for selecting whether classification or results

alejoramtor committed:
In this class are the methods for saving the results for a specific team

alejoramtor committed:
Needed for running the test

alejoramtor committed:
Classes for running and personalizing the test scenario(s)

alejoramtor committed:
Auto created classes

alejoramtor committed:
Created the class

alejoramtor committed:
Created the entity class

alejoramtor committed:
no message

alejoramtor committed:
Changed the locator strategy

alejoramtor committed:
Added checkMatchdayIsPlayed() and moveToPreviousMatchday() methods

alejoramtor committed:
Added setLanguage() method for setting spanish if the page opens in english

alejoramtor committed:
Changed the method call

alejoramtor committed:
Added method for saving the match for the entered team

alejoramtor committed:
Added the method call

alejoramtor committed:
Added class for saving the match of the desired team and calculating the score

alejoramtor committed:
Changed lightly the way some getter or setter respond

alejoramtor committed:
Added the method for moving to the classification tab

alejoramtor committed:
Added the method call for the previous matchday if the current one is not played

alejoramtor committed:
Added the instantiation of the entities and the calculation of the score

alejoramtor committed:
Updated the methods for calculating the points for the team X

alejoramtor committed:
Removed unused constructor

alejoramtor committed:
Removed unused import and attribute

alejoramtor committed:
Added this class for the classification section, getting the played games and score of team X

alejoramtor committed:
Removed unused attribute

alejoramtor committed:
Removed unused methods/atts and updated the existing ones

alejoramtor committed:
Renamed the class

alejoramtor committed:
Missed the {}

alejoramtor committed:
Updated the setLanguage method and removed unused imports

ramsesdm commented on Training2.iml:
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This file should not be under Version Control

AlejoART replied:
Done!

ramsesdm commented on src/main/java/football/entities/Match.java:
```diff
@@ -0,0 +1,86 @@
+package football.entities;
+
+public class Match {
+    String team1;
```
private

AlejoART replied:
Done!

ramsesdm commented on src/test/java/training2/football/OsasunaResultsTest.java:
```diff
@@ -0,0 +1,39 @@
+        santanderPage.goToResults();
+
+        SantanderResultsPage santanderResultsPage = new SantanderResultsPage(getDriver());
+        String[] match = santanderResultsPage.getTheCurrentMatchdayMatch("Osasuna");
```
The team name shouldn't be totally hardcoded as part of the test. Easy solution: declare a constant and use it

AlejoART replied:
Done

ramsesdm commented on src/test/java/training2/football/XTeamResultsTest.java:
```diff
@@ -0,0 +1,63 @@
+        int matchdays = santanderResultsPage.getMatchdaysAmount();
+        matches = new Match[matchdays];
+
+        for (int i = 0; i < matchdays; i++) {
```
Use foreach instead a classic for

ramsesdm commented on src/test/java/training2/football/XTeamResultsTest.java:
```diff
@@ -0,0 +1,63 @@
+            }
+        }
+        Team team = new Team();
+        for (int i = 0; i < matches.length; i++) {
```
Use foreach instead a classic for

AlejoART replied:
Done!

ramsesdm commented on src/test/java/training2/football/XTeamResultsTest.java:
```diff
@@ -0,0 +1,63 @@
+        santanderPage.goToResults();
+
+        SantanderResultsPage santanderResultsPage = new SantanderResultsPage(getDriver());
+        int matchdays = santanderResultsPage.getMatchdaysAmount();
```
Change to matchDays

AlejoART replied:
Done!

ramsesdm commented on src/test/java/training2/football/XTeamResultsTest.java:
```diff
@@ -0,0 +1,63 @@
+import org.testng.annotations.Test;
+
+public class XTeamResultsTest extends BaseTest {
+    private Match[] matches;
```
Can be converted to local variable

AlejoART replied:
Done!

ramsesdm commented on src/main/java/football/pages/SantanderResultsPage.java:
```diff
@@ -0,0 +1,79 @@
+
+        matchdaysDropdownElement = findElements(By.cssSelector("div[class='styled__DropDownContainer-sc-1yaayks-1 gECFwF']"));
+
+        String matchday = getText(matchdaysDropdownElement.get(1));
```
Not how this part works but probably you can get the number MatchDay with the size of some list instead of parsing the data

AlejoART replied:
This method will return the matchdays amount counting from the default one. For example, once you move to Santander league in this week the default match day is 5, even when there are 6 matchdays total. The ones that actually care are the one by default and the previous ones right?
Maybe I could give the matchday amount by the size of this list: matchdaysDropdownElement = findElements(By.cssSelector("div[class='styled__DropDownContainer-sc-1yaayks-1 gECFwF']"));
But I'm not sure if the amount of matchdays is always the one by default + 1

ramsesdm commented on src/main/java/football/pages/SantanderResultsPage.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    public int getMatchdaysAmount(){
+        WebDriverWait ewait1 = new WebDriverWait(driver,10);
```
WebDriverWait is already deprecated

alejoramtor committed:
Added the private to the attributes and slightly modified two methods

alejoramtor committed:
Updated the locator

alejoramtor committed:
Updated the locator

alejoramtor committed:
Updated getTheCurrentMatchdayMatch() method

alejoramtor committed:
Added the maximize() action for the driver

alejoramtor committed:
Made matches ariable local
Set the teamName as a constant
Changed the classic for loop for the foreach
Added the checkIfNull() method
